packages/adapters/src/utilities/referencedMetadataProvider.ts

```diff
@@ -0,0 +1,170 @@
+    const sopModule = metaData.get(MetadataModules.SOP_COMMON, imageId);
+    const seriesModule = metaData.get(MetadataModules.GENERAL_SERIES, imageId);
+
+    return {
+      SeriesInstanceUID: seriesModule.seriesInstanceUID,
+      ReferencedInstanceSequence: [
+        {
+          ReferencedSOPClassUID: sopModule.sopClassUID,
+          ReferencedSOPInstanceUID: sopModule.sopInstanceUID,
+        },
+      ],
+    };
+  },
+
+  /**
+   * Returns a predecessor sequence and related information to apply
+   * to the generated object.
+   */
+  [MetadataModules.PREDECESSOR_SEQUENCE]: (imageId) => {
+    // Start with the series data
+    const result = { ...metaData.get(MetadataModules.SERIES_DATA) };
+    // And extend with the predecessor information, plus updates for a new
+    // instance.
+    const generalImage = metaData.get(MetadataModules.GENERAL_IMAGE);
+    const study = metaData.get(MetadataModules.GENERAL_STUDY);
+    result.InstanceNumber = 1 + Number(generalImage.instanceNumber);
+    result.PredecessorDocumentsSequence = {
+      StudyInstanceUID: study.studyInstanceUID,
+      ReferencedSeriesSequence: {
+        SeriesInstanceUID: result.SeriesInstanceUID,
+        ReferencedSOPSequence: {
+          ReferencedSOPClassUID: generalImage.sopClassUID,
+          ReferencedSOPInstanceUID: generalImage.sopInstanceUID,
+        },
+      },
+    };
+    return result;
+  },
+
+  /**
+   * Returns a Study header instance data for a given image.
+   */
+  [MetadataModules.STUDY_DATA]: (imageId) => {
+    return metaData.getInstanceModule(imageId, STUDY_MODULES);
+  },
+  /**
+   * Returns a Series only header instance data for a given image.
+   */
+  [MetadataModules.SERIES_DATA]: (imageId) => {
+    return metaData.getInstanceModule(imageId, SERIES_MODULES);
+  },
+  /**
+   * Returns a Study header instance data for a given image.
+   */
+  [MetadataModules.IMAGE_DATA]: (imageId) => {
+    return metaData.getInstanceModule(imageId, IMAGE_MODULES);
+  },
+
+  /**
+   * RTSS Instance metadata
+   */
+  [MetadataModules.RTSS_INSTANCE_DATA]: (imageId) => {
+    const newInstanceData = metaData.get(
+      MetadataModules.NEW_INSTANCE_DATA,
+      imageId
+    );
+    return {
+      ...newInstanceData,
+      SeriesNumber: '3201',
+      StructureSetROISequence: [],
+      ROIContourSequence: [],
+      RTROIObservationsSequence: [],
+      ReferencedFrameOfReferenceSequence: [],
+      Modality: 'RTSTRUCT',
+      SOPClassUID: '1.2.840.10008.5.1.4.1.1.481.3', // RT Structure Set Storage
+      PositionReferenceIndicator: '',
+      StructureSetLabel: '',
+      StructureSetName: '',
+      StructureSetDate: DicomMetaDictionary.date(),
+      StructureSetTime: DicomMetaDictionary.time(),
+    };
+  },
+
+  [MetadataModules.NEW_INSTANCE_DATA]: (imageId) => {
+    const studyData = metaData.get(MetadataModules.STUDY_DATA, imageId);
+    return {
+      ...studyData,
+      SeriesNumber: '50000',
```

Comment:
seems arbitrary number, no? can you document this on top of this file as CONST

Reply:
It is arbitrary, added a comment indicating it is for replacement.